test/lib.bash

```diff
@@ -45,7 +45,49 @@ function run_e2e_tests {
     && return 1
 }
 
+function run_knative_serving_tests {
+  (
+  # Checkout the relevant code to run
+  mkdir -p "$GOPATH/src/knative.dev"
+  cd "$GOPATH/src/knative.dev" || exit
+  #git clone -b "release-v$1" --single-branch https://github.com/openshift/knative-serving.git serving
+  git clone -b "ocp-flag-0.9.0" --single-branch https://github.com/markusthoemmes/knative-serving.git serving
+  cd serving || exit
+
+  # Remove unneeded manifest
+  rm test/config/100-istio-default-domain.yaml
+
+  # Create test resources (namespaces, configMaps, secrets)
+  oc apply -f test/config
+  oc adm policy add-scc-to-user privileged -z default -n serving-tests
+  oc adm policy add-scc-to-user privileged -z default -n serving-tests-alt
+  # adding scc for anyuid to test TestShouldRunAsUserContainerDefault.
+  oc adm policy add-scc-to-user anyuid -z default -n serving-tests
+
+  failed=0
+  image_template="registry.svc.ci.openshift.org/openshift/knative-$1:knative-serving-test-{{.Name}}"
+  export GATEWAY_NAMESPACE_OVERRIDE="knative-serving-ingress"
+  go test -v -tags=e2e -count=1 -timeout=30m -parallel=3 ./test/e2e --resolvabledomain --kubeconfig "$KUBECONFIG" \
+    --imagetemplate "$image_template" \
+    || failed=1
+
+  go test -v -tags=e2e -count=1 -timeout=30m -parallel=3 ./test/conformance/runtime/... --resolvabledomain --kubeconfig "$KUBECONFIG" \
+    --imagetemplate "$image_template" \
+    || failed=1
+
+  go test -v -tags=e2e -count=1 -timeout=30m -parallel=3 ./test/conformance/api/... --resolvabledomain --kubeconfig "$KUBECONFIG" \
+    --imagetemplate "$image_template" \
+    || failed=1
+  
+  return $failed
+  )
+}
+
 function teardown {
+  if [[ "${OPENSHIFT_BUILD_NAMESPACE}" != "" ]]; then
```

Comment:
Small change here, that I didn't pushed yet. It should be:

```bash
if [[ -v OPENSHIFT_BUILD_NAMESPACE ]]; then
```

Previous check wasn't working in case you don't have `OPENSHIFT_BUILD_NAMESPACE` variable defined.